update/factory/accountDBSyncerContainerFactory.go

```diff
@@ -0,0 +1,144 @@
+package factory
+
+import (
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/syncer"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/update"
+	containers "github.com/ElrondNetwork/elrond-go/update/container"
+	"github.com/ElrondNetwork/elrond-go/update/genesis"
+)
+
+// ArgsNewTrieSyncersContainerFactory defines the arguments needed to create trie syncers container
+type ArgsNewAccountsDBSyncersContainerFactory struct {
+	TrieCacher         storage.Cacher
+	RequestHandler     update.RequestHandler
+	ShardCoordinator   sharding.Coordinator
+	Hasher             hashing.Hasher
+	Marshalizer        marshal.Marshalizer
+	TrieStorageManager data.StorageManager
+	WaitTime           time.Duration
+}
+
+type accountDBSyncersContainerFactory struct {
+	trieCacher         storage.Cacher
+	requestHandler     update.RequestHandler
+	container          update.AccountsDBSyncContainer
+	shardCoordinator   sharding.Coordinator
+	hasher             hashing.Hasher
+	marshalizer        marshal.Marshalizer
+	waitTime           time.Duration
+	trieStorageManager data.StorageManager
+}
+
+// NewAccountsDBSContainerFactory creates a factory for trie syncers container
+func NewAccountsDBSContainerFactory(args ArgsNewAccountsDBSyncersContainerFactory) (*accountDBSyncersContainerFactory, error) {
+	if check.IfNil(args.RequestHandler) {
+		return nil, update.ErrNilRequestHandler
+	}
+	if check.IfNil(args.ShardCoordinator) {
+		return nil, update.ErrNilShardCoordinator
+	}
+	if check.IfNil(args.TrieCacher) {
+		return nil, update.ErrNilCacher
+	}
+	if check.IfNil(args.Hasher) {
+		return nil, update.ErrNilHasher
+	}
+	if check.IfNil(args.Marshalizer) {
+		return nil, update.ErrNilMarshalizer
+	}
+	if check.IfNil(args.TrieStorageManager) {
+		return nil, update.ErrNilStorageManager
+	}
+	if args.WaitTime < time.Second {
```

Comment:
might extract time.Second in a const. Also, might wrap the next error return as to specify that the input wait time is lower than the set threshold.

Reply:
done